src/Css/Stylesheet.php

```diff
@@ -1672,7 +1650,7 @@ public function getDefaultStylesheet()
     {
         $options = $this->_dompdf->getOptions();
         $rootDir = realpath($options->getRootDir());
-        return $rootDir . self::DEFAULT_STYLESHEET;
+        return Helpers::build_url("file://", "", $rootDir, $rootDir . self::DEFAULT_STYLESHEET);
```

Comment:
Just appending `"file://"` should be enough here, no need for `build_url`.

Reply:
Okay, now I understand. Is building `$full_url` even necessary in any case?

Reply:
It's true that `explode_url` already determines the protocol is "file://" if no protocol is found. Which is really all we need here.

I guess the one use case to consider is a user manually loading a stylesheet outside the HTML. `build_url` will ensure that the path is normalized if, for example, the user were to provide a relative filesystem reference. Though even that use case looks excessive since chroot checks are made after running local files through realpath.

Reply:
Though we parse the default rootDir through `realpath` it's possible the user might modify the option. The user-specified value is not currently parsed through `realpath`. So while not optimal this is a fairly straightforward method to get the actual file path.

The way some of these default values are handled, and the ability for the user to customize, could be reworked in a future release.

Reply:
The `rootDir` option is passed through `realpath` here, directly above, no matter if the default or user-specified:

```php
$rootDir = realpath($options->getRootDir());
```

Or do you mean something else?

Reply:
No, you're right. I was looking at the Options class when I was making that comment and didn't follow the code all the way through.

I still think the current logic works. The Options class doesn't run the user-specified value through realpath. If the user modifies `rootDir` but specifies a relative path the current logic ensures that the default stylesheet path is fully normalized.

We don't need to build the URL in load_css_file since internal use of the load_css_file method is after already building the file URL. Anyone attempting to manually load CSS would just need to ensure that their URL conforms to expectations (has a protocol, or they've enabled URIs that don't specify a protocol).

Reply:
True, I guess its fine.